From 872ebaa0905c2d21eb22115c4b336187463a3de6 Mon Sep 17 00:00:00 2001
From: Guo Mengqi <guomengqi3@huawei.com>
Date: Thu, 3 Nov 2022 06:41:55 +0000
Subject: [PATCH] mm/sharepool: delete unused codes
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 872ebaa0905c2d21eb22115c4b336187463a3de6
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5QETC
CVE: NA

--------------------------------

sp_make_share_k2u only supports vmalloc address now. Therefore, delete a
backup handle case.

Also master is guaranteed not be freed until master->node_list is emptied.

Signed-off-by: Guo Mengqi <guomengqi3@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/share_pool.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 0534987fcadc..938ee9a90aeb 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -2657,12 +2657,11 @@ static int is_vmap_hugepage(unsigned long addr)
 
 static unsigned long __sp_remap_get_pfn(unsigned long kva)
 {
-	unsigned long pfn;
+	unsigned long pfn = -EINVAL;
 
+	/* sp_make_share_k2u only support vmalloc address */
 	if (is_vmalloc_addr((void *)kva))
 		pfn = vmalloc_to_pfn((void *)kva);
-	else
-		pfn = virt_to_pfn(kva);
 
 	return pfn;
 }
@@ -4073,11 +4072,6 @@ static int proc_usage_by_group(int id, void *p, void *data)
 	list_for_each_entry(spg_node, &spg->procs, proc_node) {
 
 		master = spg_node->master;
-		if (!master) {
-			pr_info("master is NULL! process %d, group %d\n",
-					spg_node->instat.tgid, id);
-			continue;
-		}
 		mm = master->mm;
 		tgid = master->instat.tgid;
 
-- 
2.33.0

